text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,631 @@
+
+## General Availability
+
+After the _Developer Preview_ has been sufficiently vetted, a timeline for _General Availability_ will be agreed upon by
+maintainers. The _General Availability_ checkist includes the following items:
+
+- [ ] Documentation
+  - [ ] API Reference Documentation
+  - [ ] User Guide
+  - [ ] Migration Guide
+  - [ ] Final maintenance plan for v1
+- [ ] Quality Assurance
+  - [ ] Complete suite of integration tests
+  - [ ] Examples built and tested against v2
+- [ ] No v1 feature is missing from v2 (unless it was deprecated in v1)
+
+Once the criteria have been satisfied, the high-level procedure for officially releasing CDK v2 is:
+
+1. Create a new `v1` branch from the `master` branch
+   1. Configure GitHub branch protection on `v1` as appropriate
+   1. Deploy CI/CD infrastructure on top of `v1`
+1. Make `next` the new `master` branch:
+   1. Force-push `next` on `master`
+   1. Delete `next` and all related CI/CD infrastructure
+1. Perform a _normal_ release
+1. Ceremoniously annouce CDK v2 is _Generally Available_, and repeat the v1 maintenance plan
+
+## Clean Up
+
+Now that the new major version has been declared _Generally Available_, the previous major (`v1`) has entered
+_maintenance_ mode. Forward porting is no longer exercised: instead, bug fixes on `master` will be back-ported to `v1`
+when necessary. This means `master` can diverge more significantly from `v1` than in earler stages of the process.
+
+The definition of `v1` feature flags and the code paths supporthing the _old behavior_ of those can be deleted (they
+were previously simply _neutralized_ in `v2` so that they could no longer be disabled there).
+
+## Monitoring Adoption
+
+We will keep an eye on adoption of CDK v2 thanks to the [version reporting] dataset. Should adoption pick up too slowly,
+actions will be taken to understand the causes for slow adoption and identify opportunities to provide additional
+tooling to remove roadblocks and facilitate migration.
+
+[version reporting]: https://docs.aws.amazon.com/cdk/latest/guide/tools.html
+
+## Deprecation of v1
+
+Based on the [maintenance policy](#aws-cdk-maintenance-policy), new feature development is to stop on CDK v1, and only
+critical bugs and security issues will be addressed in the `v1` release line.
+
+New GitHub labels will be added to differentiate bugs reported against `v1` from those reported against `v2`. Bugs
+reported against `v1` will be closed with a standard message during triaging unless they are identified as critical or
+security-impacting (meaning they qualify for `P0` handling).
+
+A similar provision applies to pull-requests, as they will only be accepted in the `v1` branch if they are non-breaking
+and fall into one of the following categories:
+
+- features dedicated to facilitating migration from `v1` to `v2`
+- features back-ported from `v2`
+- bug fixes (where relevant, the author is expected to submit the `v2` fix at the same time)
```

Comment:
```suggestion
- `P0` bug fixes (where relevant, the author is expected to submit the `v2` fix at the same time)
```

Is this correct (based on the above paragraph)? We will only accept `P0` bugs on `v1`, so we'll only accept PRs for `P0` bug fixes?

Reply:
No - I think we should accept bug fixes from the community for any severity, but we (the core team) will not author them.